File: src/main/java/com/est/curdsample/dto/TaskDescription.java
package com.est.curdsample.dto;

import com.est.curdsample.domain.Task;
import com.est.curdsample.util.PriorityResolver;
import com.est.curdsample.util.TimeFormatter;
import lombok.AllArgsConstructor;
import lombok.Data;

@Data
@AllArgsConstructor
public class TaskDescription {

    private String code;

    private String title;
    private String description;

    private Integer priority;
    private boolean completeStatus;

    private String startDate;
    private String endDate;

    private String createdAt;
    private String updatedAt;

    public static TaskDescription from(Task task) {

        return new TaskDescription(
            task.getCode(),
            task.getTitle(),
            task.getDescription(),
            task.getPriority(),
            task.isCompleteStatus(),
            TimeFormatter.convertToString(task.getStartTime()),
            TimeFormatter.convertToString(task.getEndTime()),
            TimeFormatter.convertToString(task.getCreatedAt()),
            TimeFormatter.convertToString(task.getUpdatedAt())
        );
    }

}

File: src/main/java/com/est/curdsample/dto/TaskDescription.kt
package com.est.curdsample.dto

import com.est.curdsample.domain.Task
import com.est.curdsample.util.PriorityResolver
import com.est.curdsample.util.TimeFormatter
import lombok.AllArgsConstructor
import lombok.Data

data class TaskDescription(

    val code: String? = null,

    val title: String? = null,

    val description: String? = null,

    val priority: Int? = null,

    val completeStatus: Boolean = false,

    val startDate: String? = null,

    val dueDate: String? = null,

    val createdAt: String? = null,

    val updatedAt: String? = null

) {

    val priorityLevel: String
        get() = PriorityResolver.resolve(priority)

    companion object {
        @JvmStatic
        fun from(task: Task): TaskDescription {
            return TaskDescription(
                task.code,
                task.title,
                task.description,
                task.priority,
                task.completeStatus,
                TimeFormatter.convertToString(task.startTime),
                TimeFormatter.convertToString(task.endTime),
                TimeFormatter.convertToString(task.createdAt),
                TimeFormatter.convertToString(task.updatedAt)
            )
        }
    }
}

